### PR TITLE
Update MVP's indicator queuing mechanism

### PR DESCRIPTION
**Pull request by biyeun**
- jqmq is outdated and no longer is really maintained or supported by newer versions of jquery
- using deffered objects now to set up a queue of indicators.

@snopoke cc @benrudolph 
fixes: http://manage.dimagi.com/default.asp?222506

### CODE REVIEW

File: custom/_legacy/mvp/static/mvp/js/mvp.indicatorQueue.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    self.next = function () {
+        self.qIndex ++;
+        var indicators = ko.utils.unwrapObservable(indicators);
+        if (self.qIndex < self.queueLength) {
+            fnProcessNextCallback(self.qIndex, self.next);
+        } else {
+            self.d.resolve();
+        }
+    }
```
Comment (dimagimon) [inline custom/_legacy/mvp/static/mvp/js/mvp.indicatorQueue.js]: Missing semicolon.

File: custom/_legacy/mvp/static/mvp/js/mvp.indicatorQueue.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    self.processError = function (data) {
+        self.numRetries ++;
+        if (self.numRetries <= self.maxRetries) {
+            self.getIndicator();
+        } else {
+            self.fnCallbackError();
+            self.d.fail();
+        }
+    }
```
Comment (dimagimon) [inline custom/_legacy/mvp/static/mvp/js/mvp.indicatorQueue.js]: Missing semicolon.

File: custom/_legacy/mvp/static/mvp/js/mvp.indicatorQueue.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    'use strict';
+    var self = this;
+    self.qIndex = -1;
+    self.queueLength = queueLength;
+    self.d = null;
+
+    self.start = function () {
+        self.d = $.Deferred();
+        self.d.done(function () {
+            console.log("complete");
```
Comment (snopoke) [inline custom/_legacy/mvp/static/mvp/js/mvp.indicatorQueue.js]: Console log

**Comment by snopoke** [comment]
Seems OK but I'm not that familiar with this fancy JS. could someone from @dimagi/js-team please review

File: custom/_legacy/mvp/static/mvp/ko/mvp.async_indicators.js [hunk 4]
```diff
@@ -68,40 +86,33 @@ var MVISIndicator = function (indicator) {
...
+
+    self.loadSuccess = function (data) {
         if (!data.error) {
             self.percentages(data.table.percentages || []);
             self.numerators(data.table.numerators || []);
             self.denominators(data.table.denominators || []);
             self.is_loaded(true);
             $('.mvp-table').trigger('mvp.loaded');
+        } else {
+            console.log("Retrying, ERROR: ", error);
```
Comment (snopoke) [inline custom/_legacy/mvp/static/mvp/ko/mvp.async_indicators.js]: Console log - possibly OK here but I don't see how the retry works.

Comment (biyeun) [inline custom/_legacy/mvp/static/mvp/ko/mvp.async_indicators.js]: intentional, but it should be `data.error`

**Comment by biyeun** [comment]
@millerdev might be a good candidate :)

File: custom/_legacy/mvp/static/mvp/ko/mvp.async_indicators.js (hunk 4 above)
Comment (biyeun) [inline custom/_legacy/mvp/static/mvp/ko/mvp.async_indicators.js]: also removed retry

**Comment by biyeun** [comment]
or @benrudolph ---can someone please at least look at this today? It's preventing MVP from loading their reports and this should get in asap.

**Comment by benrudolph** [comment]
:+1:

**Pull request merged**